XMonad/Hooks/DynamicLog.hs

```diff
@@ -433,32 +446,35 @@ statusBar' :: LayoutClass l Window
                            -- ^ The desired key binding to toggle bar visibility
            -> XConfig l    -- ^ The base config
            -> IO (XConfig (ModifiedLayout AvoidStruts l))
-statusBar' cmd xpp k conf = do
-    h <- spawnPipe cmd
-    makeStatusBar
-        (xpp >>= \pp -> dynamicLogWithPP pp{ ppOutput = hPutStrLn h })
-        mempty
-        k
-        conf
-
--- | Helper function to make status bars.  This should not be used on its own;
--- use 'statusBarProp' or 'statusBar' (or their respective variants) instead.
+statusBar' = makeSBWithKey .: statusBarHandleConfig'
+
+-- | Incorporates a 'StatusBarConfig' into an 'XConfig' by taking care of the
+-- necessary plumbing. This should not be used on its own; use 'statusBarProp'
+-- or 'statusBar' (or their respective variants) instead.
 makeStatusBar :: LayoutClass l Window
-              => X ()      -- ^ 'logHook' to execute
-              -> X ()      -- ^ 'startupHook' to execute
-              -> (XConfig Layout -> (KeyMask, KeySym))
-                           -- ^ The desired key binding to toggle bar visibility
-              -> XConfig l -- ^ The base config
+              => StatusBarConfig -- ^ The status bar config
+              -> XConfig l       -- ^ The base config
               -> IO (XConfig (ModifiedLayout AvoidStruts l))
-makeStatusBar lh sh k conf = pure $ docks $ conf
+makeStatusBar (StatusBarConfig lh sh ch k action) conf = pure $ docks $ conf
```

Comment:
I thought about doing that, but isn't normally the `XConfig l` the last parameter? Or it doesn't matter?

Reply:
It's normally the last parameter for ergonomics reason; you can simply compose things nicer that way.  This breaks down here because `makeStatusBar` is used in a sort of "unintended" way (where we already have the config already, but we want to change something with the status bar options), so having the `XConfig` argument first here would make more sense